main.js

```diff
@@ -27,11 +27,11 @@ window.iitcBuildDate = '@@BUILDDATE@@';
 window.onload = function() {};
 document.body.onload = function() {};
 
-
 //originally code here parsed the <Script> tags from the page to find the one that defined the PLAYER object
 //however, that's already been executed, so we can just access PLAYER - no messing around needed!
 
-if (typeof(window.PLAYER)!="object" || typeof(window.PLAYER.nickname) != "string") {
+var PLAYER = window.PLAYER || (unsafeWindow && unsafeWindow.PLAYER);
```

Comment:
```suggestion
var PLAYER = window.PLAYER || (typeof unsafeWindow !== 'undefined' && unsafeWindow.PLAYER);
```
Otherwise we have TypeError when `window.PLAYER` is not defined (which is normal when user is not logged in yet).